commit b1332c51133c591602493cbcdf5fb01f6a1ef601 (refs/changes/43/5043/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-11-09T10:50:19-08:00 (11 months ago)
    
    TRITON-945 vm-agent tests broken by TRITON-810 (3 missing updates)
    Reviewed by: Dave Eddy <dave.eddy@joyent.com>
    Approved by: Dave Eddy <dave.eddy@joyent.com>

diff --git a/tests/test.VmWatcher.js b/tests/test.VmWatcher.js
index 2822bd1..2c1000f 100644
--- a/tests/test.VmWatcher.js
+++ b/tests/test.VmWatcher.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 var execFile = require('child_process').execFile;
@@ -61,7 +61,8 @@ test('load existing VMs', function _test(t) {
 
 test('determine best event source', function _test(t) {
     var opts = {
-        log: mocks.Logger
+        log: mocks.Logger,
+        vmadm: vmadm
     };
 
     determineEventSource(opts,
@@ -81,7 +82,8 @@ test('starting VmWatcher', function _test(t) {
     watcher = new VmWatcher({
         log: mocks.Logger,
         eventSource: eventSource,
-        periodicInterval: PERIODIC_INTERVAL
+        periodicInterval: PERIODIC_INTERVAL,
+        vmadm: vmadm
     });
 
     t.ok(watcher, 'created VmWatcher');
diff --git a/tests/test.VmadmEventsWatcher.js b/tests/test.VmadmEventsWatcher.js
index 7e1bbf9..49a3356 100644
--- a/tests/test.VmadmEventsWatcher.js
+++ b/tests/test.VmadmEventsWatcher.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 var execFile = require('child_process').execFile;
@@ -57,7 +57,8 @@ function main() {
 
         watcher = new VmadmEventsWatcher({
             log: mocks.Logger,
-            updateVm: _onVmUpdate
+            updateVm: _onVmUpdate,
+            vmadm: vmadm
         });
 
         t.ok(watcher, 'created VmadmEventsWatcher');
@@ -150,7 +151,8 @@ function main() {
 
 test('determine best event source', function _test(t) {
     var opts = {
-        log: mocks.Logger
+        log: mocks.Logger,
+        vmadm: vmadm
     };
 
     determineEventSource(opts,
diff --git a/tests/test.ZoneeventWatcherOverflow.js b/tests/test.ZoneeventWatcherOverflow.js
index c96fe07..f141005 100644
--- a/tests/test.ZoneeventWatcherOverflow.js
+++ b/tests/test.ZoneeventWatcherOverflow.js
@@ -265,7 +265,8 @@ function main() {
 
 test('determine best event source', function _test(t) {
     var opts = {
-        log: mocks.Logger
+        log: mocks.Logger,
+        vmadm: vmadm
     };
 
     determineEventSource(opts,
